### DISCUSSION TITLE
Using homebrew to distribute strawberry ?

### DISCUSSION THREAD

**Discussion by phwuil** [strawberrymusicplayer/strawberry#1317] [Q&A] [answered]
Hi,

For mac users, what about distributing strawberrymusicplayer using homebrew ?

**Comment by phwuil** [comment]
Hi, for now, trying to compile it using homebrew, after installing a large bunch of packages, I am blocked here 

> ICU_ROOT=/opt/homebrew/Cellar/icu4c/73.2 cmake ..
[...]
KDSingleApplication/src/kdsingleapplication.cpp

CMake Error at 3rdparty/kdsingleapplication/CMakeLists.txt:5 (add_library):
  No SOURCES given to target: kdsingleapplication


Any clue ?

**Reply by phwuil** [reply, marked as answer]
ok, its a submodule :-)

**Comment by davidbitterlich** [comment]
Hi, unfortunatelly the author of this great music player decided to provide binaries only to paying sponsors and I understand some of the reasons he mentioned in his post on the strawberry forum: https://forum.strawberrymusicplayer.org/topic/1308/access-to-macos-downloads

However, I would also love to see a homebrew package of strawberry. Compilation on macOS is pretty challenging, especially: install qt6 via brew, then build everything, then remove libraries from qt package itself (I don't really understand why. Is it because of linking? Or is it because broken libraries are added to the strawberry package?) and then reinstall qt6 before the next compilation of strawberry will be performed... Also I was unable to manage the build to have no issues in deploycheck. However, the binary works in my environment...

I'm not sure about homebrew, but is it still possible to deploy unsigned apps using it? If so, one could manually sign afterwards: codesign --deep --force -s - strawberry.app

**Comment by jonaski** [comment]
There was a homebrew package of strawberry, but it was removed when I restricted the macos dmgs to sponsors.
We no longer use homebrew to build strawberry. There was simply too many issues using homebrew for building strawberry, and I do not like the way homebrew run their project. They also simply refused to enable simple things like gstreamer mpg123 plugin for MP3 playback, so it falls back to using libav which isn't as good. I also had to wait with a new strawberry release every time there was something broken with homebrew or an upstream bug requiring a patch, sometimes patches takes months to get in.
I simply do not want to have to deal with homebrew ever again.

**Reply by phwuil** [reply]
Hi @davidbitterlich and @jonaski, thanks for your answers!

@jonaski, too bad to hear that. I am just a homebrew user and like it a lot. But I understand your concerns.

However, I have to tell that I succeed to compile strawberry almost flawlessly. Every time cmake notified me that something was missing, I just installed it with homebrew. I even installed some optional dependencies. The biggest issue was me searching for KDSingleApplication everywhere except as a submodule (as you can see above) :-) 

I did not even have to sign the app. 
It is not perfect (maybe because some libraries are still missing), but it is already by far the best music player for mac. 

Thanks a lot !

PS- I did not have to remove anything from qt before running strawberry. 

Here are the lines for compiling strawberry :
The version numbers  73.2 and 25.1 will certainly have to change from time to time ...

```
mkdir build
cd build
ICU_ROOT=/opt/homebrew/Cellar/icu4c/73.2 cmake ..
CPATH=/opt/homebrew/Cellar/protobuf/25.1/include make -j
```
cmake will show you what to install. At the end, when everything's fine, you have produced the file straberry.app
